### PR TITLE
Ticket #6 Fetch Single Article

### PR DESCRIPTION
**Pull request by Skahrr**

### CODE REVIEW

File: src/components/ArticleDisplay.jsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+export const ArticleDisplay = ({ article }) => {
+  return (
+    <article>
+      <div className="title"><Link to={`/articles/${article.topic}/${article.article_id}`}>{article.title}</Link></div>
+      <div className="information">
+        <div className="author">Written by: {article.author}</div>
+        <div className="created_at">{article.created_at}</div>
+        <div className="comment_count">Comments: {article.comment_count}</div>
+      </div>
```
Comment (douglashellowell) [inline src/components/ArticleDisplay.jsx:11]: `<div>` tags are great for grouping and styling elements but shouldn't be used for just containing text

There are better HTML tags to use for text like `<p>` tags

File: src/components/SingleArticle.jsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  }, [article_id]);
+  {console.log(article);}
+  return (
+    <main>
+      <div className="title-single-article">{article.title}</div>
+      <div className="article-body">{article.body}</div>
+      <div className="information">
+        <div className="author">Written by: {article.author}</div>
+        <div className="created_at">{article.created_at}</div>
+        <div className="comment_count">Comments: {article.comment_count}</div>
```
Comment (douglashellowell) [inline src/components/SingleArticle.jsx:23]: more `<div>`s here to replace...

File: src/components/SingleArticle.jsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+export const SingleArticle = () => {
+  let { article_id } = useParams();
+  const [article, setArticle] = useState([]);
+  useEffect(() => {
+    fetchSingleArticle(article_id).then((article) => {
+      setArticle(article);
+    });
+  }, [article_id]);
+  {console.log(article);}
```
Comment (douglashellowell) [inline src/components/SingleArticle.jsx:15]: be sure to remove logs before commiting 👀

File: src/App.js [hunk 4]
```diff
@@ -17,6 +18,7 @@ function App() {
             <Route path="/" element={<Navigate to="/articles" />} />
             <Route path="/articles/:topic" element={<Main />} />
             <Route path="/articles" element={<Main />} />
+            <Route path="/articles/:topic/:article_id" element={<SingleArticle/>}/>
```
Comment (douglashellowell) [inline src/App.js:21]: this isn't a huge problem but be aware that `/articles/not-a-topic/1` will successfully render article 1!

Thankfully your back button still sends to the correct topic page

**Comment by douglashellowell** [comment]
looks good!

a few issues i've mentioned in comments above ☝🏻 

---

With a slow connection it takes time for your articles to appear and users with poor signal only see a blank screen while this happens!
Think about what you want your users to see while the data is loading
You might need to check the conditional rendering notes for this :book:
(this goes for all api requests throughout your app)

---

base functionality is here so feel free to merge when ready! 💪🏻

**Comment by Skahrr** [comment]
I'll try and fix all that before continuing to the next ticket
Thank you very much!!

**Pull request merged**